packages/contracts/contracts/TellerV2.sol

```diff
@@ -801,12 +812,33 @@ contract TellerV2 is
         // Start with the original due date being 1 payment cycle since bid was accepted
         dueDate_ = bid.loanDetails.acceptedTimestamp + bid.terms.paymentCycle;
 
+        // Calculate due date if payment cycle is set to monthly
+        if (bidPaymentCycleType[_bidId] == PaymentCycleType.Monthly) {
+            dueDate_ = uint32(
+                BokkyPooBahsDateTimeLibrary.addMonths(
+                    bid.loanDetails.acceptedTimestamp,
+                    1
+                )
+            );
+        }
+
         // Calculate the cycle number the last repayment was made
-        uint32 delta = lastRepaidTimestamp(_bidId) -
-            bid.loanDetails.acceptedTimestamp;
+        uint32 lastRepaidTimestamp = lastRepaidTimestamp(_bidId);
+        uint32 delta = lastRepaidTimestamp - bid.loanDetails.acceptedTimestamp;
         if (delta > 0) {
             uint32 repaymentCycle = 1 + (delta / bid.terms.paymentCycle);
-            dueDate_ += (repaymentCycle * bid.terms.paymentCycle);
+            if (bidPaymentCycleType[_bidId] == PaymentCycleType.Monthly) {
+                dueDate_ = uint32(
+                    BokkyPooBahsDateTimeLibrary.addMonths(
+                        dueDate_,
+                        repaymentCycle
```

Comment:
this is the wrong delta calculation for `Monthly` - `repaymentCycle` needs to calculate the number of actual _months_ between the dates by using the `BokkyPooBahsDateTimeLibrary`